### ISSUE TITLE
Wrong CJK font in the UI

### ISSUE THREAD

**Issue by Z-Clive** [zealdocs/zeal#536]
Hello guys,

Thanks for this amazing work. 

I'm currently using the latest version of `Zeal` on `Windows 10` and the default font in the list and other stuff like settings panel is not so good. Hopefully you could implement a feature to allow users change the default font shows inside the app.

**Comment by trollixx** [comment]
Zeal should use the font your system provides, but I do see the problem on your screenshot in #537.

**Comment by ZClive** [comment]
I thought so, for `Zeal` shows the default font (aka. `Segoe UI`) in the list pane. However, either in the search result pane or in the list pane where the item was truncated, it shows a different font. Here's an example (the example for the search results are shown in #537 ):

![2016_04_20_10_33_09_mousegesturel](https://cloud.githubusercontent.com/assets/11990819/14661881/de369092-06e3-11e6-810d-36b1581f30fd.png)

**Comment by trollixx** [comment]
Could you check if the latest [CI build](https://bintray.com/zealdocs/windows-ci/zeal-unstable/view) has the same problem?

**Comment by ZClive** [comment]
@trollixx Yep, those 2 problems seem to be fixed in the latest CI Build. Although that ugly font still shows in few places like settings panel, it's much more usable and pretty now.

**Comment by trollixx** [comment]
Where exactly in settings? Available/Installed docset lists?

**Comment by ZClive** [comment]
@trollixx Here's another screenshot to demonstrate such issue:

![1](https://cloud.githubusercontent.com/assets/11990819/14674027/27572d7c-0734-11e6-8f6f-40021af21fef.png)

**Comment by trollixx** [comment]
Very strange. What is your Windows language? Any changes to default font or
theme settings?

**Comment by ZClive** [comment]
The language is `zh-CN`. And absolutely no modification to default font or theme.

**Comment by trollixx** [comment]
@Z-Clive There were several improvements in the related code, could you give a CI build another try?

**Comment by bosbyj** [comment]
This is a bug of long history from QT, that messes up with Chinese/Japanese/Korean fonts. Unless Zeal presets the UI font globally as san-serif/serif, there is no quick remedy.

**Comment by trollixx** [comment]
Indeed seems like a bug in Qt, but I am not sure how to debug this, considering I don't have Chinese Windows.

@liangqi I saw people talking to you on IRC regarding Qt issues on Chinese Windows. Any ideas about what could be wrong here?

**Comment by bosbyj** [comment]
It has been a QT bug for cjk fonts for decades. And from how the app looks,
you can tell if it is made with QT or other technologies.

You have to override the font config manually.

**Comment by trollixx** [comment]
Considering that in certain places fonts are rendered properly, it feels like the bug should be easy to fix.

**Comment by trollixx** [comment]
@bosbyj @Z-Clive Could you give Zeal v0.5.0 a try and see if the problem is exactly the same? I think the screenshots above are from v0.3.1, and there were some changes since then.

**Comment by bosbyj** [comment]
I call this a "SimSun Ghost in QT".

Back in the old days, when people were using Windows 95, 98, XP, Simsun was the prevailing font set in simplifed Chinese operation systems. QT3/QT4 would detect this font and put it into default use.

However, even if you changed the default font system wide, for example, to TAHOMA, some parts of QT's widget still got rendered by Simsun font type. This also occured in the later Windows Vista, 7, 8, 10, while the simplified Chinese version OS chose Segoe or MS Yahei as its default font.

A mixture of the real system default font and "Simsun".

Respectively, it could also happen in Japanese Windows, with the Mincho and Gothic fonts.

![image](https://user-images.githubusercontent.com/7922167/34639127-7cb274f4-f2b8-11e7-8ad8-100353a0a5d5.png)

![image](https://user-images.githubusercontent.com/7922167/34639129-864de2dc-f2b8-11e7-8a45-f3ebd6ce0097.png)

**Comment by trollixx** [comment]
@bosbyj Thanks for the background story.

It looks like parts rendered by the application itself (list delegates) are fine, and all default Qt's rendering is off. Considering we have a lot of Chinese users, I would love to have this fixed, but I need to find some time and dig into how Qt chooses fonts. This seems like a font priority bug, or Qt not reading the system preference by default.

**Comment by bosbyj** [comment]
You are welcome and thank you for your effort.

I am user of many QT based gui apps, to name a few, GoldenDict, Anki, Calibre ... and they all look alike, with font messed up by Simsun. I now take this a feature of QT, trying to avoid perfectionism.

If you want to fix it, you may ship an open source cjk font and override the setting globally for all Chinese-Japanese-Korean users.

**Comment by trollixx** [comment]
I would rather prefer to fix this in Qt. And it's definitely possible since we already have some parts of the application rendered properly. I'll try to find some time to figure out why the wrong font gets selected.

**Comment by bosbyj** [comment]
Thank you. For your information on Japanese font, same scenario after
windows 7.

https://bugreports.qt.io/browse/QTBUG-54760

**Comment by trollixx** [comment]
Found [this SO thread](https://stackoverflow.com/q/7043904), that basically says that using SimSun is the correct behavior.

Also, can anyone try setting locale? Just add the following line in the very beginning of `[main()](https://github.com/zealdocs/zeal/blob/84afbf51483c5144f77c0cfe4b482fb450c23a99/src/app/main.cpp#L175)`.

```cpp
QLocale::setDefault(QLocale::C);

// or

QLocale::setDefault(QStringLiteral("en_US"));
```

**Comment by trollixx** [comment]
Closing this. It should be fixed in Qt 6, which we've shipped for a while now.

The cause was Qt 5 picking a Latin-only default font (MS Shell Dlg, substituted to Tahoma) for the UI. With no CJK glyphs, Windows filled them in from its font-link chain, where SimSun comes first, hence the ugly fallback in native widgets while our own list delegates looked fine.

Qt 6.0 switched to the proper localized system font (YaHei UI on Chinese, Meiryo on Japanese, etc.), so the SimSun fallback no longer kicks in. See [QTBUG-58610](https://qt-project.atlassian.net/browse/QTBUG-58610) (fixed in 6.0). The original reporter on [QTBUG-87391](https://qt-project.atlassian.net/browse/QTBUG-87391) confirmed it fixed in a 6.0 build. That ticket just stayed open over an unrelated Qt 5 backport question.

If anyone still sees the wrong font on a Qt 6 build, please post an update with a screenshot and your Windows display language, and I'll reopen.

<details>
<summary>Note to self: when a font override would actually be needed</summary>

The Qt 6 fix only covers the UI-chrome font, chosen for the Windows **display language**. An explicit `QApplication::setFont` + `setFamilies` override is only worth adding if we need to draw text in a script the UI-language font doesn't cover:

- **Non-CJK Windows showing CJK text** (e.g. English UI, Chinese docset names or search input). Segoe UI has no CJK glyphs, so Windows font-links them and can still land on SimSun.
- **Mixed CJK** (e.g. Simplified-Chinese Windows running YaHei UI but rendering Japanese/Korean). Han unification gives wrong regional glyph shapes.

Not relevant for: matching CJK Windows (this issue, fixed in 6.0), or the docs pages themselves (rendered by QtWebEngine/Chromium, separate fallback). Windows 10/11 also reordered the font-link chain to prefer YaHei/Yu Gothic over SimSun, so even the residual case is rarer now. Only add the override if a real mismatched-locale report shows up.
</details>
